src/pudl/transform/params/ferc1.py

```diff
@@ -1779,6 +1779,7 @@
                     "ElectricExpensesSteamPowerGeneration": "opex_electric",
                     "MaintenanceOfStructuresSteamPowerGeneration": "opex_structures",
                     "ReportYear": "report_year",
+                    "entity_id": "entity_id",
```

Comment:
`assign_record_id` now effectively requires having this line in the rename dictionary for xbrl. I didn't like this because it is not actually required right now *for the renaming* and thus is mixing what this rename dict is used for without any warning or communication.

Reply:
I suppose with the id regiggering needed for #1705 we will actually want to rename both the `entity_id` and the `respondent_id` to `utility_id_ferc1_xbrl` and `utility_id_ferc1_dbf` so this secret dependency won't exist.

Reply:
There's a warning in `rename_columns()` isn't there? If there's any discrepancy between the columns in the dataframe being renamed, and the keys in the rename dictionary you get a `logger.warning()` telling you so and reporting what all columns were not shared by the two sets.  That's why I was asking whether you thought that ought to be an actual exception, rather than just a logger warning.

Reply:
But I agree that accessing `rename_columns` for the purpose of constructing the list of primary keys is an off-label use, which is likely to be confusing and result in unexpected coupling and problems down the line.

Reply:
I guess it feels weird to force `rename_columns` to know all of the columns, even the ones we will drop/you don't actually need to rename. So I appreciate the `logger.warning` bc it is informative but that is not really my problem - the coupling is my problem.

I think this particular problem is going to be resolved when we use and rename `entity_id`

Reply:
I think the real problem here is that we're attempting to use `rename_columns` for more than one thing -- both to rename the columns (its stated purpose) -- and as a source of metadata about the incoming raw dataframe, with some implicit rules about how to interpret the columns it finds to turn them into e.g. primary keys.

I think we'll be able to avoid this mess entirely when we've merged in the PR from @zschira that outputs the full metadata / table schemas that are associated with the SQLite DBs feeding the tables into this process. Then `rename_columns` can just be for renaming and if we need the other explicit structural information we can just grab it from the real metadata.